src/main/java/edu/cornell/kfs/paymentworks/util/PaymentWorksNewVendorConversionUtil.java

```diff
@@ -405,9 +382,71 @@ public PaymentWorksVendor createPaymentWorksNewVendor(
 	}
 
+	protected void extractCustomFields(PaymentWorksNewVendorDetailDTO paymentWorksNewVendorDetailDTO,
+			PaymentWorksUtil paymentWorksUtil, PaymentWorksVendor paymentWorksNewVendor) {
+		Map<String, String> customFieldsMap = paymentWorksUtil.convertFieldArrayToMap(paymentWorksNewVendorDetailDTO.getCustom_fields());
+		logCustomFieldsFromPaymentWorks(customFieldsMap);
+		for (String payamentWorkCustomFieldName : customFieldsMap.keySet()) {
+			String customFieldValueFromPaymentWorks = customFieldsMap.get(payamentWorkCustomFieldName);
+			
+			PaymentWorksFieldMapping fieldMapping = findPaymentWorksFieldMapping(payamentWorkCustomFieldName);
+			
+			if (ObjectUtils.isNotNull(fieldMapping)) {
+				Object propertyValueForSetter = convertStringValueToObjectForPropertySetting(customFieldsMap.get(payamentWorkCustomFieldName));
+				if (ObjectUtils.isNotNull(propertyValueForSetter)) {
+					try {
+						ObjectUtils.setObjectProperty(paymentWorksNewVendor, fieldMapping.getKfsFieldName(), propertyValueForSetter);
+						if (LOG.isDebugEnabled()) {
+							LOG.debug("extractCustomFields setting KFS field " + fieldMapping.getKfsFieldName() + " with a value of " + 
+									propertyValueForSetter.toString() + " and is of object type " + propertyValueForSetter.getClass());
+						}
+					} catch (FormatException | IllegalAccessException | InvocationTargetException
+							| NoSuchMethodException e) {
+						LOG.error("extractCustomFields Unable to set kfs field '" + fieldMapping.getKfsFieldName() + "' with a value of '" + 
+							propertyValueForSetter.toString() + "'  Due to the error: " +  e.getMessage());
+					}
+				}
+				
+			} else {
+				LOG.error("extractCustomFields, Unable to find a KFS field for the payment works field of '" + payamentWorkCustomFieldName + KRADConstants.SINGLE_QUOTE);
+			}
+		}
+	}
+	
+	private void logCustomFieldsFromPaymentWorks(Map<String, String> customFieldsMap) {
+		if (LOG.isDebugEnabled()) {
+			for (String key : customFieldsMap.keySet()) {
+				String val = customFieldsMap.get(key);
+				LOG.debug("logCustomFieldsFromPaymentWorks  key:'" + key + "'  value: '" + val + KRADConstants.SINGLE_QUOTE);
+			}
+		}
+	}
+
+	protected PaymentWorksFieldMapping findPaymentWorksFieldMapping(String payamentWorkCustomFieldName) {
+		Map fieldValues = new HashMap();
+		fieldValues.put("PMW_FIELD", StringUtils.trim(payamentWorkCustomFieldName));
+		Collection mappings = getBusinessObjectService().findMatching(PaymentWorksFieldMapping.class, fieldValues);
+		if (!mappings.isEmpty()) {
+			return (PaymentWorksFieldMapping)mappings.iterator().next();
+		}
+		return null;
+	}
+	
+	protected Object convertStringValueToObjectForPropertySetting(String value) {
+		if (StringUtils.equalsIgnoreCase(value, "YES") || StringUtils.equalsIgnoreCase(value, "NO")) {
+			Boolean calculatedBooleanValue = StringUtils.equalsIgnoreCase(value, "YES");
+			return calculatedBooleanValue;
+		}
+		if (StringUtils.isBlank(value)) {
+			return null;
```

Comment:
I was specifically using NULL.  The function is only used once, and does a NULL check before it tries to use reflection to set a field.